xbmc/guilib/GUIComponent.cpp

```diff
@@ -35,13 +39,27 @@ CGUIComponent::~CGUIComponent()
 {
   CServiceBroker::UnregisterGUI();
   m_pWindowManager->DeInitialize();
+  
+  m_pWindowManager.reset();
+  m_pTextureManager.reset();
+  m_pLargeTextureManager.reset();
```

Comment:
Can you deinitialize these in reverse order?

Reply:
if this matters, something is really wrong and we should fix the root cause.

Reply:
If we reverse them, it will show us if anything is wrong so we can fix it.

Reply:
deinit is reverse order is happy path. doing it differently challenges the system more.

Reply:
huh? isn't fixing the system worth the effort?

Reply:
Yes ofc, this is why changing the order here should not matter. Keeping the order in a sequence that is worse, will bring issues to surface faster

Reply:
I don't understand, isn't RAII-inspired order, where resources are unacquired in the opposite order, better than what is here now?

Reply:
Right, uninit in reverse order is better in general. But the order should not matter here. If it matters I want to know and fix the actual issue.

Reply:
It matters from a maintainability standpoint. It's easier for a programmer to verify the order is reversed properly than try to reason about the intention of breaking convention.

Reply:
Ok, then I will remove those lines completely. The only reason they are pointers is to avoid includes in header file

Reply:
Nonexistent code is easiest to maintain :)